core/src/test/scala/org/apache/spark/sql/delta/DeltaOptionSuite.scala

```diff
@@ -177,4 +179,62 @@ class DeltaOptionSuite extends QueryTest
       }
     }
   }
+
+  test("support no compression write option (defaults to snappy)") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+      withTable("compression") {
+        spark.range(100)
+          .writeTo("compression")
+          .using("delta")
+          .option("compression", "snappy")
+          .tableProperty("location", path)
+          .create()
+        assert(FileUtils.listFiles(tempDir, Array("snappy.parquet"), false).size > 0)
+      }
+    }
+  }
+
+  test("support all parquet compression codecs as write option (except LZO and BROTLI)") {
+    val ignore = Set("lzo", "brotli")
+    val codecs = CompressionCodec.values()
+      .map(_.name.toLowerCase(Locale.ROOT))
+      .filterNot(ignore.contains)
+    val subExtensions = Array("", "snappy.", "gz.", "lz4.", "zstd.")
+    assert(subExtensions.size === codecs.size)
+
+    codecs.zip(subExtensions).foreach { case (codec, subExt) =>
```

Comment:
if you are zipping `codecs` with a static array `subExtensions`, why not just statically define `codecs`? this would make it more explicit and clear to the reader exactly what codec is being paired with which subExtensions

Reply:
could you also change this from one test case ("test all except XY") into multiple test cases? this would make it more clear which test is failing, if one ever were to
e.g. 
```scala
my_test_tuples.foreach { case (a, b) =>
  test("support compression code ...") {

  }
}
```

Reply:
code rewritten